platforms/common/work_queue/hrt_thread.c

```diff
@@ -138,6 +138,14 @@ static void hrt_work_process()
 	/* Default to sleeping for 1 sec */
 	next  = 1000000;
 
+#ifdef __PX4_QURT
+	// With Qurt we have to wake up more often because there is an assumption
+	// that certain signals wake up a sleeping thread but it isn't the case
+	// with the Qurt POSIX implementation. So rather than assume we can come out
+	// of the sleep early by a signal we just wake up more often.
+	next = 1000;
```

Comment:
How about we just drop the signals entirely and instead of sleep wait on a semaphore (with a timeout for the next item)?

Reply:
Okay, I'll take a look